scripts/benchmarks_monthly.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+'''
+This script provides simple execution of the .NET micro benchmarks
+for a .NET preview release, making it easier to contribute to our
+monthly manual performance runs.
+'''
+
+from argparse import ArgumentParser, ArgumentTypeError
+from logging import getLogger
+
+import benchmarks_ci
+import datetime
+import tarfile
+import sys
+import os
+
+VERSIONS = {
+    'net7.0-preview2': { 'tfm': 'net7.0', 'build': '7.0.100-preview.2.22124.4' },
+    'net7.0-preview1': { 'tfm': 'net7.0', 'build': '7.0.100-preview.1.22077.12' },
+    'net6.0': { 'tfm': 'net6.0' }
+}
+
+def get_version_from_name(name: str) -> str:
+    for version in VERSIONS:
+        if version == name:
+            return VERSIONS[version]
+
+    raise Exception('The version specified is not supported', name)
+
+def add_arguments(parser: ArgumentParser) -> ArgumentParser:
+    '''Adds new arguments to the specified ArgumentParser object.'''
+
+    if not isinstance(parser, ArgumentParser):
+        raise TypeError('Invalid parser.')
+
+    parser.add_argument(
+        'version',
+        choices=VERSIONS,
+        help='Specifies the .NET release for the benchmarks run')
+
+    parser.add_argument(
+        '--filter',
+        dest='filter',
+        default='*',
+        help='Specifies the benchmark filter to pass to BenchmarkDotNet')
+
+    parser.add_argument(
+        '--architecture',
+        dest='architecture',
+        required=False,
+        choices=['x64', 'x86', 'arm64', 'arm'],
+        default='x64',
+        help='Specifies the processor architecture to pass to BenchmarkDotNet')
+
+    parser.add_argument(
+        '--bdn-arguments',
+        dest='bdn_arguments',
+        required=False,
+        help='Command line arguments to be passed to BenchmarkDotNet, wrapped in quotes',
+    )
+
+    return parser
+
+def __process_arguments(args: list):
+    parser = ArgumentParser(
+        description='Tool to execute the monthly manual micro benchmark performance runs',
+        allow_abbrev=False
+    )
+
+    add_arguments(parser)
+    return parser.parse_args(args)
+
+def __main(args: list) -> int:
+    args = __process_arguments(args)
+
```

Comment:
if run with no arguments, it could emit some example command line(s) perhaps?

Reply:
eh, I guess it's pretty obvious, and most people will be pasting something from an email anyway